packages/ncids-js/src/components/nci-header/nci-megamenu.component.ts

```diff
@@ -0,0 +1,561 @@
+		}
+
+		this.initialize();
+		NCIMegaMenu._components.set(this.element, this);
+	}
+
+	/**
+	 * Instantiates this component of the given element.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @param {Partial<NCIMegaMenuOptions>} options Component being created.
+	 * @public
+	 */
+	public static create(
+		element: HTMLElement,
+		options: NCIMegaMenuOptions
+	): NCIMegaMenu {
+		if (!(element instanceof HTMLElement)) {
+			throw 'Element is not an HTMLElement';
+		}
+		return this._components.get(element) || new this(element, options);
+	}
+
+	/**
+	 * Resets component and primary navigation
+	 * to a clean state before the init()
+	 *
+	 * @public
+	 */
+	public unregister(): void {
+		this.removeToggleEventListeners();
+		const sections = this.queryListSections();
+		sections.forEach((section) => {
+			const selector = '.nci-header-nav__primary-button';
+			const button = section.querySelector(selector);
+			const menu = this.queryMegaMenu(<HTMLElement>button);
+			menu.remove();
+			this.deactivateButton(<HTMLElement>button);
+		});
+		document.removeEventListener('click', this.eventListener, false);
+		document.removeEventListener('keydown', this.keyboardListener, false);
+		NCIMegaMenu._components.delete(this.element);
+	}
+
+	/**
+	 * Sets up component by initializing the items
+	 * 1. Create MegaMenu layers and Buttons
+	 * 2. Find new buttons with tags add actions
+	 * 3. Custom event listners
+	 * 4. Attach event listners for document (a11y)
+	 * @private
+	 */
+	private initialize(): void {
+		this.createMegaMenuSections();
+		this.findToggleButtons();
+		this.createCustomEvents();
+		document.addEventListener('click', this.eventOffListener, false);
+		document.addEventListener('keydown', this.keyboardListener, false);
+	}
+
+	/**
+	 * Create MegaMenu section based on Primary
+	 * navigation links.
+	 *
+	 * @private
+	 */
+	private createMegaMenuSections(): void {
+		// fin all LI elements in navigation
+		const sections = this.queryListSections();
+		sections.forEach((section) => {
+			// find link elements and convert to button
+			const selector = this.options.linkElementClass;
+			const link = section.querySelectorAll(<string>selector)[0];
+			const id = link.getAttribute('href') || '';
+			const parentLabel = link.innerHTML;
+			// add aria tags and change link to button
+			this.activateButton(link);
+
+			// dom creation for menu layers
+			const menu = this.createDom(
+				'div',
+				['nci-header-megamenu', 'hidden'],
+				[{ 'aria-expanded': 'false' }, { id: id }]
+			);
+
+			// parent Row for MegaMenu
+			const gridRow = this.createDom('div', ['grid-row']);
+			// col and Section Header for MegaMenu
+			const gridColHeader = this.createDom('div', ['usa-col', 'grid-gap-3']);
+			const sectionHeader = this.createDom('div', [
+				'nci-header-megamenu__header',
+			]);
+			// explore Section Link
+			const sectionLink = this.createDom(
+				'a',
+				[],
+				[{ href: '/path/here' }, { id: id }]
```

Comment:
can we link this somewhere? is this the href of the original link in the navbar?

Reply:
error - should be id (as id is path of original link and whats used for api) - updating.